src/Request/Validator/UpdateLineItemValidator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Request\Validator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Throwable;
+
+class UpdateLineItemValidator
+{
+    /** @var ValidatorInterface */
+    private $validator;
+
+    public function __construct(ValidatorInterface $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    public function validate(Request $request): void
+    {
+        try {
+            $responseBody = json_decode($request->getContent(), true, 512, JSON_THROW_ON_ERROR);
+        } catch (Throwable $jsonException) {
+            throw new BadRequestHttpException(
+                sprintf(
+                    'Invalid JSON request body received. Error: %s.',
+                    $jsonException->getMessage()
+                ),
+                $jsonException
+            );
+        }
+
+        $errors = $this->validator->validate($responseBody, $this->getConstraints());
+
+        if ($errors->count() < 1) {
+            return;
+        }
+
+        $errorsRaw = [];
+        /** @var ConstraintViolationInterface $error */
+        foreach ($errors as $error) {
+            $errorsRaw[] = sprintf(
+                "%s -> %s",
+                $error->getPropertyPath(),
+                $error->getMessage()
+            );
+        }
+
+        throw new BadRequestHttpException(
+            sprintf(
+                'Invalid Request Body: %s',
+                implode(
+                    " ",
+                    $errorsRaw
+                )
```

Comment:
We don't need multiple lines for the implode section. Also maybe separate it with comma